master/internal/rm/kubernetesrm/pods.go

```diff
@@ -934,99 +948,103 @@ func (p *pods) handleGetAgentsRequest(ctx *actor.Context) {
+			// If all of a node's taints are tolerated by a pool, that node belongs to the pool.
+			if allTaintsTolerated(node.Spec.Taints, poolTolerations) {
+				poolsToNodes[poolName] = append(poolsToNodes[poolName], node)
+			}
 		}
+	}
 
-		// If there's no quota for our only resource pool's namespace
-		if _, ok := namespaceToQuota[namespaceOfPool]; !ok {
-			return p.summarizeClusterByNodes(ctx), nil
-		}
+	f := logrus.Fields{
+		"lenPoolsToNodes":        len(poolsToNodes),
+		"lenNamespaceToPoolName": len(p.namespaceToPoolName),
+	}
+	for _, pool := range poolsToNodes {
+		key := fmt.Sprintf("lenNodesIn-%s", pool)
+		f[key] = len(poolsToNodes["default"])
 	}
+	logrus.WithFields(f).Debug("done associating nodes and pools")
 
+	// Build the set of summaries for each resource pool
 	containers := p.containersPerResourcePool()
 	summaries := make(map[string]model.AgentSummary, len(p.namespaceToPoolName))
-	for namespace, poolName := range p.namespaceToPoolName {
+	for poolName, nodes := range poolsToNodes {
 		slots := model.SlotsSummary{}
-		numContainers := containers[poolName]
-		var registeredTime time.Time
-		if quota, quotaExists := namespaceToQuota[namespace]; quotaExists {
-			slots = make(map[string]model.SlotSummary)
-			registeredTime = quota.CreationTimestamp.Time
-
-			for resourceName, qty := range quota.Spec.Hard {
-				var deviceType device.Type
-				switch resourceName {
-				case k8sV1.ResourceCPU:
-					deviceType = device.CPU
-				case ResourceTypeNvidia, "limits." + ResourceTypeNvidia:
-					deviceType = device.CUDA
-				default:
-					// We only care about CPU and GPU quotas for the slots summary
-					continue
-				}
+		numContainersInPool := containers[poolName]
 
-				// Each CPU and GPU in the quota will be counted as a slot here
-				one, decQty := inf.NewDec(1, 0), qty.AsDec()
-				for i := inf.NewDec(0, 0); i.Cmp(decQty) < 0; i.Add(i, one) {
-					id := fmt.Sprintf("%s/%s/%s", poolName, string(deviceType), i.String())
-
-					var container *cproto.Container
-					// Create a number of pseudo-containers in the summary equal to the number of
-					// running containers
-					if decNumContainers := inf.NewDec(int64(numContainers),
-						0); i.Cmp(decNumContainers) < 0 {
-						container = &cproto.Container{
-							ID:    cproto.ID(id),
-							State: "RUNNING",
-						}
-					}
+		// We'll create a number of pseudo-containers in the summary equal to the number of
+		// running containers in this pool.
+		pseudoContainersAdded := 0
+
+		for _, node := range nodes {
+			numSlots, slotType := extractSlotInfo(nodeSummaries[node.Name])
 
-					slots[id] = model.SlotSummary{
-						ID:        id,
-						Device:    device.Device{Type: deviceType},
-						Enabled:   true,
-						Container: container,
+			for j := 0; j < numSlots; j++ {
+				id := fmt.Sprintf("%s/%s/%s/%d", poolName, node.Name, string(slotType), j)
+
+				var container *cproto.Container
+				if pseudoContainersAdded < numContainersInPool {
```

Comment:
do we need to increment `pseudoContainersAdded`?

Reply:
Good catch!